File: CustomTableView/CustomTableView/ViewController.m
//
//  ViewController.m
//  CustomTableView
//
//  Created by Nguyen Hung Thanh Liem on 1/16/19.
//  Copyright © 2019 Nguyen Hung Thanh Liem. All rights reserved.
//

#import "ViewController.h"
#import "SimpleTableCell.h"

@interface ViewController ()
@property (weak, nonatomic) IBOutlet UITableView *tableViewMain;

@end

@implementation ViewController
{
    NSArray *tableData;
    NSArray *tableImage;
    NSArray *tableTime;
}

- (void)viewDidLoad {
    [super viewDidLoad];
    tableData = [NSArray arrayWithObjects:@"Egg Benedict", @"Mushroom Risotto", @"Full Breakfast", @"Hamburger", @"Ham and Egg Sandwich", @"Creme Brelee", @"White Chocolate Donut", @"Starbucks Coffee", @"Vegetable Curry", @"Instant Noodle with Egg", @"Noodle with BBQ Pork", @"Japanese Noodle with Pork", @"Green Tea", @"Thai Shrimp Cake", @"Angry Birds Cake", @"Ham and Cheese Panini", nil];
    
    tableImage = [NSArray arrayWithObjects:@"gradient_bilinear.jpg",@"gradient_fx_spherical.gif",@"images.jpeg",@"size100.jpg",@"gradient_bilinear.jpg",@"gradient_fx_spherical.gif",@"images.jpeg",@"size100.jpg",@"gradient_bilinear.jpg",@"gradient_fx_spherical.gif",@"images.jpeg",@"size100.jpg",@"gradient_bilinear.jpg",@"gradient_fx_spherical.gif",@"images.jpeg",@"size100.jpg", nil];
    tableTime = [NSArray arrayWithObjects:@"time",@"time",@"time",@"time",@"time",@"time",@"time",@"time",@"time",@"time",@"time",@"time",@"time",@"time",@"time",@"time",@"time", nil];

}
//Methods

- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section
{
    return [tableData count];
}

- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath
{
    static NSString *simpleTableIdentifier = @"SimpleTableCell";
    
    SimpleTableCell *cell = (SimpleTableCell *)[tableView dequeueReusableCellWithIdentifier:simpleTableIdentifier];
    if (cell == nil)
    {
        NSArray *nib = [[NSBundle mainBundle] loadNibNamed:@"SimpleTableCell" owner:self options:nil];
        cell = [nib objectAtIndex:0];
    }
    
    cell.nameLabel.text = [tableData objectAtIndex:indexPath.row];
    cell.thumbnailImageView.image = [UIImage imageNamed:[tableImage objectAtIndex:indexPath.row]];
    cell.prepTimeLabel.text = [tableTime objectAtIndex:indexPath.row];
    
    return cell;
}
- (CGFloat)tableView:(UITableView *)tableView heightForRowAtIndexPath:(NSIndexPath *)indexPath
{
    return 78;
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}


@end

File: TableView-Swift/TableView-Swift/ViewController.swift
//
//  ViewController.swift
//  TableView-Swift
//
//  Created by Nguyen Hung Thanh Liem on 1/17/19.
//  Copyright © 2019 Nguyen Hung Thanh Liem. All rights reserved.
//

import UIKit

class ViewController: UIViewController,UITableViewDataSource,UITableViewDelegate {
    
    
    @IBOutlet weak var myTable: UITableView!
    var myarray:[String]!
    
    override func viewDidLoad() {
        super.viewDidLoad()
        myTable.dataSource=self
        myarray=["Android","PHP","IOS","Ruby"]
        
    }

    //TableView methods


    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        return myarray.count
    }
    
    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
        //TableCell
        let cell = tableView.dequeueReusableCell(withIdentifier: "CELL") as! DONGTableViewCell
        //Set view data
        cell.imgHinh.image=UIImage (named: "images")
        cell.myLabel.text=myarray[indexPath.row]
        return cell
    }
    func tableView(_ tableView: UITableView, heightForRowAt indexPath: IndexPath) -> CGFloat {
        return 200
    }
    
    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }


}

